applications/StructuralMechanicsApplication/python_scripts/structural_response.py

```diff
@@ -374,11 +374,10 @@ def _SynchronizeAdjointFromPrimal(self):
             adjoint_node.Z = primal_node.Z
 
         # Put primal solution on adjoint model - for "auto" setting, else it has to be done by the user e.g. using hdf5 process
-        if self.response_settings["adjoint_settings"].GetString() == "auto":
-            Logger.PrintInfo("> Transfer primal state to adjoint model part.")
-            variable_utils = KratosMultiphysics.VariableUtils()
-            for variable in self.primal_state_variables:
-                variable_utils.CopyModelPartNodalVar(variable, self.primal_model_part, self.adjoint_model_part, 0)
+        Logger.PrintInfo("> Transfer primal state to adjoint model part.")
```

Comment:
Why do you removed the if "auto" check? I case one uses hdf5 the primal solution is written twice to the adjoint model part. Or I am wrong?

Reply:
You are right. The problem is, that if somebody specifies the adjoint parameters without the "auto" feature (only because the output of a special nodal variable is needed), then one is also forced to specify an HDF5 in- and output process, which has further, significant impacts (compiled HDF5-APP). On the other hand, doing twice the job is also bad. I am very open to suggestions to solve this problem.